src/qml/qgismobileapp.qml

```diff
@@ -1559,6 +1571,51 @@ ApplicationWindow {
             }
           }
         }
+
+        QfToolButton {
+          id: curveEditionButton
+          width: visible ? 40 : 0
+          height: visible ? 40 : 0
+          padding: 2
+          round: true
+          visible: dashBoard.activeLayer && (dashBoard.activeLayer.geometryType() === Qgis.GeometryType.Polygon || dashBoard.activeLayer.geometryType() === Qgis.GeometryType.Line)
```

Comment:
I think we want this to be dependent on `QgsWkbTypes.isCurvedType()`

Reply:
In my example , my layer is not a curve type at all and i use QgsCompoundCurve::curveToLine function to set geometry drawn into my none curved layer.  This helps me to drawn faster instead of set vertex by vertex.

But you are right i am not taking into account the possibility to set directly the qgscompoundcurve geometry into the new feature.

How do you think ?

Reply:
Interesting, I was mostly seeing this in the context of surveying / cadastre, where curves are often a requirement. My main motivation here would be to keep the interface as light as possible and ask users to have a curve layer if they want this feature. What do you think?

Reply:
I am using it to draw underground maps, such as caves or old quarries. My polygon layer corresponds to some galleries and i no need to have a specific curved layer but some times digitizing arcs like in qgis help me.

Maybe we can allow it only for curved layer (QgsWkbTypes.isCurvedType()) or for advanced digitizing user ( which could be set in settings).

Tell me what you think ?

Sorry for the delay, i had some stuff to do last week.

Reply:
Interesting job you have!

I'd be interested to hear @nirvn's point of view

Reply:
@m-kuhn , @qsavoye , hey there, sorry for the delay in chipping in. 

I do think the feature is useful; I would indeed prefer if this option was only visible for curved type layers, I feel that'd be a good way to enable this while not cluttering UI by limiting to an explicit intent by the user / project manager.